src/main/kotlin/com/terraformation/backend/tracking/db/ObservationResultsStore.kt

```diff
@@ -75,8 +83,38 @@ class ObservationResultsStore(private val dslContext: DSLContext) {
     return fetchByCondition(OBSERVATIONS.plantingSites.ORGANIZATION_ID.eq(organizationId), limit)
   }
 
-  fun fetchSummaryForPlantingSite(
+  /**
+   * Retrieves historical summaries for a planting site by aggregating most recent observation data
+   * per subzone, ordered chronologically, latest first. Each summary represents the summary data
+   * after an observation is completed.
+   *
+   * @param limit if provided, only return this number of historical summary.
+   * @param maxCompletionTime if provided, only observations completed before will be used.
+   */
+  fun fetchSummariesForPlantingSite(
+      plantingSiteId: PlantingSiteId,
+      limit: Int? = null,
+      maxCompletionTime: Instant? = null,
+  ): List<ObservationRollupResultsModel> {
+    val completedObservations =
+        fetchByPlantingSiteId(plantingSiteId, maxCompletionTime = maxCompletionTime).filter {
+          it.completedTime != null
+        }
+
+    val numObservations = completedObservations.size
+    val depth = max(0, limit?.let { min(it, numObservations) } ?: numObservations)
+
+    // Remove observations one at a time, to build a historical summary
+    return List(depth) {
+          val observations = completedObservations.takeLast(numObservations - it)
```

Comment:
`.drop(it)` would make the intent a little clearer than `.takeLast(numObservations - it)`.